docs/docs/administer/upgrade.md

```diff
@@ -25,6 +25,40 @@ To upgrade Firezone, follow these steps:
 Occasionally problems arise. If you hit any, please let us know by [filing an
 issue](https://github.com/firezone/firezone/issues/new/choose).
 
+## Upgrading from 0.3.x to >= 0.3.16
+
+Follow the instructions below based on your current version and setup:
+
+### I have an existing OIDC integration
+
+You will need to repeat the OIDC setup steps to
+[create a new app integration]({%link docs/authenticate/index.md%}).
+
+Upgrading to >= 0.3.16 requires the `offline_access` scope for OIDC integrations.
+This ensures Firezone syncs with the identity provider and VPN access is terminated
+once the user is removed.
+
+The refresh token required to sync users with your identity
+provider can only be obtained the first time a user grants permissions to Firezone.
+So even if the scope was provided previously, you will need to re-setup the OIDC
+integration after upgrading.
```

Comment:
How does one re-setup the OIDC integration? I think we may get some confused users here. Either the Oauth app needs to be completely removed and recreated (new client_id and client_secret) or the user needs to revoke the permissions for the app (I think this varies on a case-by-case).

Option (1) is the one that should work in most cases.